src/pages/search/SearchTypeDropdown.tsx

```diff
@@ -0,0 +1,70 @@
+import NotesIcon from '@mui/icons-material/Notes';
+import PersonIcon from '@mui/icons-material/Person';
+import ShowChartIcon from '@mui/icons-material/ShowChart';
+import { MenuItem, TextField, TextFieldProps } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { dataTestId } from '../../utils/dataTestIds';
+
+export enum SearchTypeValue {
+  Result = 'registration',
+  Person = 'person',
+  Project = 'project',
+}
+
+interface SearchTypeDropdownProps extends Pick<TextFieldProps, 'sx'> {
+  selectedValue: SearchTypeValue;
+  onSelectResult: () => void;
+  onSelectPerson: () => void;
+  onSelectProject: () => void;
+}
+
+export const SearchTypeDropdown = ({
+  sx = {},
+  selectedValue,
+  onSelectResult,
+  onSelectPerson,
+  onSelectProject,
+}: SearchTypeDropdownProps) => {
+  const { t } = useTranslation();
+
+  return (
+    <TextField
+      select
+      value={selectedValue}
+      size="small"
+      data-testid={dataTestId.startPage.searchTypeField}
+      sx={{
+        minWidth: '9rem',
+        '.MuiSelect-select': {
+          display: 'flex',
+          gap: '0.5rem',
+          alignItems: 'center',
+          bgcolor: `${selectedValue}.main`,
+        },
+        ...sx,
+      }}
+      slotProps={{ htmlInput: { 'aria-label': t('common.type') } }}>
+      <MenuItem
+        sx={{ display: 'flex', gap: '0.5rem', alignItems: 'center' }}
```

Comment:
Når man har lik styling flere ganger som for hvert MenuItem her, så kan det forenkle litt å flytte stylingen til en styled component man legger over/utenfor komponenten:

```
const StyledMenuItem = styled(MenuItem)({
  display: 'flex',
  gap: '0.5rem',
  alignItems: 'center',
});
```

```suggestion
      <StyledMenuItem
```